core/src/trezor/crypto/bech32.py

```diff
@@ -125,6 +126,68 @@ def convertbits(
 
     return ret
 
+def bech32_decode(bech32_str):
+    # if not (1 <= len(bech32_str) <= 90):
+    #     raise ValueError("Invalid length of Bech32 string")
+
+    # if not all(char.lower() in CHARSET for char in bech32_str):
+    #     raise ValueError("Invalid characters in Bech32 string")
+
+    # if not (bech32_str.lower() == bech32_str or bech32_str.upper() == bech32_str):
+    #     raise ValueError("Mixed case in Bech32 string")
+
+    # if "1" not in bech32_str:
+    #     raise ValueError("Missing separator in Bech32 string")
+
+    # if bech32_str.index("1") == 0 or bech32_str.index("1") == len(bech32_str) - 1:
+    #     raise ValueError("Invalid position of separator in Bech32 string")
+
+    hrp, data = bech32_str.rsplit("1", 1)
+    hrp = hrp.lower()
+    data = data.lower()
+
+    # for char in hrp:
+    #     if char not in CHARSET:
+    #         raise ValueError("Invalid character in HRP")
+
+    for char in data:
+        if char not in CHARSET:
+            raise ValueError("Invalid character in data part")
+
+    decoded = []
+    for char in data:
+        decoded.append(CHARSET_REV[char])
+
+    # if not bech32_verify_checksum(hrp, decoded):
+    #     raise ValueError("Invalid checksum")
+
+    return hrp, decoded[:-6]
+
+def reverse_convertbits(
+    data: Sequence[int], frombits: int, tobits: int
+) -> list[int]:
+    acc = 0
+    bits = 0
+    ret = []
+    maxv = (1 << frombits) - 1
+    for value in data:
+        if value < 0 or (value >> tobits):
+            raise ValueError  # input value does not match `tobits` size
+        acc = ((acc << tobits) | value) & ((1 << (frombits + tobits)) - 1)
+        bits += tobits
+        while bits >= frombits:
+            bits -= frombits
+            ret.append((acc >> bits) & maxv)
+    return ret
```

Comment:
Why is this needed? As I can see, the existing `convertbits` can convert in both directions, i.e. 8 to 5 and 5 to 8. Am I missing something?

Reply:
it is not working correctly in the other way.